packages/puppeteer-core/src/cdp/ExecutionContext.ts

```diff
@@ -50,8 +57,91 @@ export class ExecutionContext {
     if (contextPayload.name) {
       this._contextName = contextPayload.name;
     }
+    this._client.on('Runtime.bindingCalled', this.#onBindingCalled);
   }
 
+  // Set of bindings that have been registered in the current context.
+  #contextBindings = new Set<string>();
+
+  // Contains mapping from functions that should be bound to Puppeteer functions.
+  #bindings = new Map<string, Binding>();
+
+  // If multiple waitFor are set up asynchronously, we need to wait for the
+  // first one to set up the binding in the page before running the others.
+  #mutex = new Mutex();
+  async _addBindingToContext(name: string): Promise<void> {
```

Comment:
Nit: we only use it in this class now. 
```suggestion
  async #addBindingToContext(name: string): Promise<void> {
```

Reply:
I will refactor internals later